### ISSUE TITLE
`balance_probabilities` is maybe wrong?

### ISSUE THREAD

**Issue by LeoGrin** [PriorLabs/TabPFN#122]
The `balance_probabilities` argument from `TapPFNClassifier` is implemented like this:
```
        if self.balance_probabilities:
            class_prob_in_train = self.class_counts_ / self.class_counts_.sum()
            output = output * torch.Tensor(class_prob_in_train).to(self.device_)
            output = output / output.sum(dim=-1, keepdim=True)
```
I'm wondering whether `output = output * torch.Tensor(class_prob_in_train).to(self.device_)` shouldn't be `output = output / torch.Tensor(class_prob_in_train).to(self.device_)` to push models toward to minority class instead of the majority class. I may be confused about what this argument is for. @eddiebergman I think you implemented this, WTYT?

**Comment by eddiebergman** [comment]
Based on what I can see in the original repos, this is likely a bug on my behalf and my guess I was thinking about it in terms of normalization. The original code seemed to look like this:

```python
    def _reweight_probs_based_on_train(
        self,
        eval_ys: torch.Tensor,
        output: torch.Tensor,
        num_classes: int,
        device: str,
    ) -> torch.Tensor:
        """
        Reweight class probabilities based on training set.

        Parameters:
            eval_ys (torch.Tensor): Label tensor for evaluation.
            output (torch.Tensor): Output probability tensor.
            num_classes (int): Number of classes.
            device (str): Computing device.

        Returns:
            torch.Tensor: Reweighted output tensor.
        """
        # make histogram of counts of each class in train set
        train_class_probs = torch.zeros(num_classes, device=device)
        train_class_probs.scatter_add_(
            0,
            eval_ys.flatten().long().to(device),
            torch.ones_like(eval_ys.flatten(), device=device),
        )
        train_class_probs = (
            train_class_probs / train_class_probs.sum()
        )  # shape: num_classes
        output /= train_class_probs
        # make sure outputs last dim sums to 1
        output /= output.sum(dim=-1, keepdim=True)

        return output
```

Feel free to change as you see fit

**Comment by LeoGrin** [comment]
Thanks @eddiebergman !! Opened a PR
